### PR TITLE
Cache longer but check load date to invalidate when needed

### PR DESCRIPTION
**Pull request by vincentlepot**
# Description

This PR optimizes the Navitia cache. Caching the results longer, but invalidating when new navitia data is loaded (check on last_load_at in coverage information)

## Pull Request type

This is a performance enhancement

## How to test

Here are the following steps to test this pull request:

- It should act the same, but cache should expire if the data in Navitia are updated

## Team reviewers

BOT Team

### CODE REVIEW

**Comment by vincentlepot** [comment]
Finally, it's not a good idea to use last_load_date because it depends on the kraken instance used. dataset_created_at?

**Comment by vincentlepot** [comment]
Seen with alex : publication_date in status. I'll do the modifications

**Comment by vincentlepot** [comment]
Jenkins retest this please

File: chaos/navitia.py [hunk 1]
```diff
@@ -93,10 +101,29 @@ def get_pt_object(self, uri, object_type, pt_objects=None):
...
-                return json[self.collections[object_type]][0]
+                rv = json[self.collections[object_type]][0]
+                cache.set(cache_key, rv, app.config['CACHE_CONFIGURATION'].get('NAVITIA_CACHE_TIMEOUT', 3600))
+                return rv
+
+        return None
+
+    @cache.memoize(timeout=app.config['CACHE_CONFIGURATION'].get('NAVITIA_PUBDATE_CACHE_TIMEOUT', 600))
+    def get_coverage_publication_date(self):
+        logging.getLogger().debug("Cache miss for publication coverage {}".format(self.coverage))
```
Comment (kadhikari) [inline chaos/navitia.py]: "Cache miss for publication coverage {}"  -> "Cache miss for publication date on coverage {}"

**Comment by kadhikari** [comment]
I didn't find another way to mock call for Publication_date

File: chaos/navitia.py (hunk 1 above)
Comment (vincentlepot) [inline chaos/navitia.py]: I fixed this message

**Comment by vincentlepot** [comment]
I fixed your feedback. I guess this PR can be merged ;)

File: chaos/navitia.py [hunk 2]
```diff
@@ -75,8 +75,16 @@ def _navitia_caller(self, query):
             # currently we reraise the previous exceptions
             raise exceptions.NavitiaError('call to navitia failed, data : {}'.format(query))
 
-    @cache.memoize(timeout=app.config['CACHE_CONFIGURATION'].get('NAVITIA_CACHE_TIMEOUT', 3600))
     def get_pt_object(self, uri, object_type, pt_objects=None):
+        cache_key = 'Chaos.get_pt_object.{}.{}.{}.{}'.format(self.get_coverage_publication_date(), uri, object_type, pt_objects)
```
Comment (kinnou02) [inline chaos/navitia.py]: It will be safer to include the coverage in the cache key

**Comment by kadhikari** [comment]
jenkins test this please

File: chaos/navitia.py (hunk 2 above)
Comment (vincentlepot) [inline chaos/navitia.py]: It's already in the URL but I agree it will be safer

**Comment by kadhikari** [comment]
I rebased the branch master with notification_date. Welcome for the remarks..

**Pull request merged**